### ISSUE TITLE
In rails console, tab completion doesn't work, and instead get DEPRECATION WARNING:

### ISSUE THREAD

**Issue by s-katoh** [rails/rails#39598]
### Steps to reproduce

As well as #37775 I am still experiencing  this issuse. 
### Expected behavior

ActiveRecord::Base.new.meth => Tab (completion should work)
### Actual behavior

ActiveRecord::Base.new.meth => Tab 
DEPRECATION WARNING: SourceAnnotationExtractor is deprecated! Use Rails::SourceAnnotationExtractor instead. (called from require at bin/rails:4)
### System configuration
**Rails version**:
6.0.3
**Ruby version**:
2.7.1
<details>
  <summary>Gemfile</summary>
  
  ```ruby '2.7.1'

source 'https://rubygems.org'

git_source(:github) do |repo_name|
  repo_name = "
  "https://github.com/
end


gem 'rails', '~> 6.0'
gem 'puma'
gem 'sass-rails'
gem 'uglifier', '>= 1.3.0'
gem 'jquery-rails'
gem 'jbuilder'
gem 'haml-rails'
gem 'dragonfly'
gem 'yaml_db'
gem 'fastercsv'
gem 'devise'
gem 'cancancan'
gem 'pg'
gem 'sprockets'
gem 'babel-transpiler'
gem 'sprockets-rails'
gem 'xmlrpc' 
gem 'gravatar-ultimate'
gem "autoprefixer-rails"
gem 'rounding'
gem 'webpacker'
gem 'excelout', git: 'git@github.com:mk-system/excelout-ruby.git'
gem 'slack-incoming-webhooks'

group :development, :test do
  
  gem 'byebug', platform: :mri
  gem 'rspec-rails'
  gem 'guard'
  gem 'guard-rspec'
  gem 'factory_bot_rails'
  gem 'capybara'
  gem 'better_errors'
  gem 'binding_of_caller'
  gem 'rack-mini-profiler'
  gem 'pry-rails'
  gem 'rubocop'
  gem 'rubocop-rails'
  gem 'rspec_junit_formatter'
end

group :development do
  
  gem 'web-console'
  gem 'listen'
  
  gem 'spring'
  gem 'spring-watcher-listen'
end

group :test do
  gem 'selenium-webdriver'
  gem 'webdrivers'
end

gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]

```
</details>

<details>
<summary>Gemfile.lock</summary>

```
GIT
  remote: git@github.com:mk-system/excelout-ruby.git
  revision: 7fb673d7786f4288a00d8a9917e32cc37c44f7bb
  specs:
    excelout (0.1.0)

GEM
  remote: https://rubygems.org/
  specs:
    actioncable (6.0.3)
      actionpack (= 6.0.3)
      nio4r (~> 2.0)
      websocket-driver (>= 0.6.1)
    actionmailbox (6.0.3)
      actionpack (= 6.0.3)
      activejob (= 6.0.3)
      activerecord (= 6.0.3)
      activestorage (= 6.0.3)
      activesupport (= 6.0.3)
      mail (>= 2.7.1)
    actionmailer (6.0.3)
      actionpack (= 6.0.3)
      actionview (= 6.0.3)
      activejob (= 6.0.3)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (6.0.3)
      actionview (= 6.0.3)
      activesupport (= 6.0.3)
      rack (~> 2.0, >= 2.0.8)
      rack-test (>= 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.2.0)
    actiontext (6.0.3)
      actionpack (= 6.0.3)
      activerecord (= 6.0.3)
      activestorage (= 6.0.3)
      activesupport (= 6.0.3)
      nokogiri (>= 1.8.5)
    actionview (6.0.3)
      activesupport (= 6.0.3)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.1, >= 1.2.0)
    activejob (6.0.3)
      activesupport (= 6.0.3)
      globalid (>= 0.3.6)
    activemodel (6.0.3)
      activesupport (= 6.0.3)
    activerecord (6.0.3)
      activemodel (= 6.0.3)
      activesupport (= 6.0.3)
    activestorage (6.0.3)
      actionpack (= 6.0.3)
      activejob (= 6.0.3)
      activerecord (= 6.0.3)
      marcel (~> 0.3.1)
    activesupport (6.0.3)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 0.7, < 2)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
      zeitwerk (~> 2.2, >= 2.2.2)
    addressable (2.7.0)
      public_suffix (>= 2.0.2, < 5.0)
    ast (2.4.0)
    autoprefixer-rails (9.7.6)
      execjs
    babel-source (5.8.35)
    babel-transpiler (0.7.0)
      babel-source (>= 4.0, < 6)
      execjs (~> 2.0)
    bcrypt (3.1.13)
    better_errors (2.7.0)
      coderay (>= 1.0.0)
      erubi (>= 1.0.0)
      rack (>= 0.9.0)
    bindex (0.8.1)
    binding_of_caller (0.8.0)
      debug_inspector (>= 0.0.1)
    builder (3.2.4)
    byebug (11.1.3)
    cancancan (3.1.0)
    capybara (3.32.1)
      addressable
      mini_mime (>= 0.1.3)
      nokogiri (~> 1.8)
      rack (>= 1.6.0)
      rack-test (>= 0.6.3)
      regexp_parser (~> 1.5)
      xpath (~> 3.2)
    childprocess (3.0.0)
    coderay (1.1.2)
    concurrent-ruby (1.1.6)
    crass (1.0.6)
    debug_inspector (0.0.3)
    devise (4.7.1)
      bcrypt (~> 3.0)
      orm_adapter (~> 0.1)
      railties (>= 4.1.0)
      responders
      warden (~> 1.2.3)
    diff-lcs (1.3)
    dragonfly (1.2.0)
      addressable (~> 2.3)
      multi_json (~> 1.0)
      rack (>= 1.3)
    erubi (1.9.0)
    erubis (2.7.0)
    execjs (2.7.0)
    factory_bot (5.2.0)
      activesupport (>= 4.2.0)
    factory_bot_rails (5.2.0)
      factory_bot (~> 5.2.0)
      railties (>= 4.2.0)
    fastercsv (1.5.5)
    ffi (1.12.2)
    formatador (0.2.5)
    globalid (0.4.2)
      activesupport (>= 4.2.0)
    gravatar-ultimate (2.0.0)
      activesupport (>= 2.3.14)
      rack
    guard (2.16.2)
      formatador (>= 0.2.4)
      listen (>= 2.7, < 4.0)
      lumberjack (>= 1.0.12, < 2.0)
      nenv (~> 0.1)
      notiffany (~> 0.0)
      pry (>= 0.9.12)
      shellany (~> 0.0)
      thor (>= 0.18.1)
    guard-compat (1.2.1)
    guard-rspec (4.7.3)
      guard (~> 2.1)
      guard-compat (~> 1.1)
      rspec (>= 2.99.0, < 4.0)
    haml (5.1.2)
      temple (>= 0.8.0)
      tilt
    haml-rails (2.0.1)
      actionpack (>= 5.1)
      activesupport (>= 5.1)
      haml (>= 4.0.6, < 6.0)
      html2haml (>= 1.0.1)
      railties (>= 5.1)
    html2haml (2.2.0)
      erubis (~> 2.7.0)
      haml (>= 4.0, < 6)
      nokogiri (>= 1.6.0)
      ruby_parser (~> 3.5)
    i18n (1.8.2)
      concurrent-ruby (~> 1.0)
    jaro_winkler (1.5.4)
    jbuilder (2.10.0)
      activesupport (>= 5.0.0)
    jquery-rails (4.3.5)
      rails-dom-testing (>= 1, < 3)
      railties (>= 4.2.0)
      thor (>= 0.14, < 2.0)
    listen (3.2.1)
      rb-fsevent (~> 0.10, >= 0.10.3)
      rb-inotify (~> 0.9, >= 0.9.10)
    loofah (2.5.0)
      crass (~> 1.0.2)
      nokogiri (>= 1.5.9)
    lumberjack (1.2.4)
    mail (2.7.1)
      mini_mime (>= 0.1.1)
    marcel (0.3.3)
      mimemagic (~> 0.3.2)
    method_source (1.0.0)
    mimemagic (0.3.5)
    mini_mime (1.0.2)
    mini_portile2 (2.4.0)
    minitest (5.14.0)
    multi_json (1.14.1)
    nenv (0.3.0)
    nio4r (2.5.2)
    nokogiri (1.10.9)
      mini_portile2 (~> 2.4.0)
    notiffany (0.1.3)
      nenv (~> 0.1)
      shellany (~> 0.0)
    orm_adapter (0.5.0)
    parallel (1.19.1)
    parser (2.7.1.2)
      ast (~> 2.4.0)
    pg (1.2.3)
    pry (0.13.1)
      coderay (~> 1.1)
      method_source (~> 1.0)
    pry-rails (0.3.9)
      pry (>= 0.10.4)
    public_suffix (4.0.4)
    puma (4.3.3)
      nio4r (~> 2.0)
    rack (2.2.2)
    rack-mini-profiler (2.0.1)
      rack (>= 1.2.0)
    rack-proxy (0.6.5)
      rack
    rack-test (1.1.0)
      rack (>= 1.0, < 3)
    rails (6.0.3)
      actioncable (= 6.0.3)
      actionmailbox (= 6.0.3)
      actionmailer (= 6.0.3)
      actionpack (= 6.0.3)
      actiontext (= 6.0.3)
      actionview (= 6.0.3)
      activejob (= 6.0.3)
      activemodel (= 6.0.3)
      activerecord (= 6.0.3)
      activestorage (= 6.0.3)
      activesupport (= 6.0.3)
      bundler (>= 1.3.0)
      railties (= 6.0.3)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.3.0)
      loofah (~> 2.3)
    railties (6.0.3)
      actionpack (= 6.0.3)
      activesupport (= 6.0.3)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.20.3, < 2.0)
    rainbow (3.0.0)
    rake (13.0.1)
    rb-fsevent (0.10.4)
    rb-inotify (0.10.1)
      ffi (~> 1.0)
    regexp_parser (1.7.0)
    responders (3.0.0)
      actionpack (>= 5.0)
      railties (>= 5.0)
    rexml (3.2.4)
    rounding (1.0.1)
    rspec (3.9.0)
      rspec-core (~> 3.9.0)
      rspec-expectations (~> 3.9.0)
      rspec-mocks (~> 3.9.0)
    rspec-core (3.9.2)
      rspec-support (~> 3.9.3)
    rspec-expectations (3.9.1)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.9.0)
    rspec-mocks (3.9.1)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.9.0)
    rspec-rails (4.0.0)
      actionpack (>= 4.2)
      activesupport (>= 4.2)
      railties (>= 4.2)
      rspec-core (~> 3.9)
      rspec-expectations (~> 3.9)
      rspec-mocks (~> 3.9)
      rspec-support (~> 3.9)
    rspec-support (3.9.3)
    rspec_junit_formatter (0.4.1)
      rspec-core (>= 2, < 4, != 2.12.0)
    rubocop (0.82.0)
      jaro_winkler (~> 1.5.1)
      parallel (~> 1.10)
      parser (>= 2.7.0.1)
      rainbow (>= 2.2.2, < 4.0)
      rexml
      ruby-progressbar (~> 1.7)
      unicode-display_width (>= 1.4.0, < 2.0)
    rubocop-rails (2.5.2)
      activesupport
      rack (>= 1.1)
      rubocop (>= 0.72.0)
    ruby-progressbar (1.10.1)
    ruby_parser (3.14.2)
      sexp_processor (~> 4.9)
    rubyzip (2.3.0)
    sass-rails (6.0.0)
      sassc-rails (~> 2.1, >= 2.1.1)
    sassc (2.3.0)
      ffi (~> 1.9)
    sassc-rails (2.1.2)
      railties (>= 4.0.0)
      sassc (>= 2.0)
      sprockets (> 3.0)
      sprockets-rails
      tilt
    selenium-webdriver (3.142.7)
      childprocess (>= 0.5, < 4.0)
      rubyzip (>= 1.2.2)
    semantic_range (2.3.0)
    sexp_processor (4.14.1)
    shellany (0.0.1)
    slack-incoming-webhooks (0.3.0)
    spring (2.1.0)
    spring-watcher-listen (2.0.1)
      listen (>= 2.7, < 4.0)
      spring (>= 1.2, < 3.0)
    sprockets (4.0.0)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.2.1)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    temple (0.8.2)
    thor (1.0.1)
    thread_safe (0.3.6)
    tilt (2.0.10)
    tzinfo (1.2.7)
      thread_safe (~> 0.1)
    uglifier (4.2.0)
      execjs (>= 0.3.0, < 3)
    unicode-display_width (1.7.0)
    warden (1.2.8)
      rack (>= 2.0.6)
    web-console (4.0.2)
      actionview (>= 6.0.0)
      activemodel (>= 6.0.0)
      bindex (>= 0.4.0)
      railties (>= 6.0.0)
    webdrivers (4.3.0)
      nokogiri (~> 1.6)
      rubyzip (>= 1.3.0)
      selenium-webdriver (>= 3.0, < 4.0)
    webpacker (5.1.1)
      activesupport (>= 5.2)
      rack-proxy (>= 0.6.1)
      railties (>= 5.2)
      semantic_range (>= 2.3.0)
    websocket-driver (0.7.1)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.4)
    xmlrpc (0.3.0)
    xpath (3.2.0)
      nokogiri (~> 1.8)
    yaml_db (0.7.0)
      rails (>= 3.0)
      rake (>= 0.8.7)
    zeitwerk (2.3.0)

PLATFORMS
  ruby

DEPENDENCIES
  autoprefixer-rails
  babel-transpiler
  better_errors
  binding_of_caller
  byebug
  cancancan
  capybara
  devise
  dragonfly
  excelout!
  factory_bot_rails
  fastercsv
  gravatar-ultimate
  guard
  guard-rspec
  haml-rails
  jbuilder
  jquery-rails
  listen
  pg
  pry-rails
  puma
  rack-mini-profiler
  rails (~> 6.0)
  rounding
  rspec-rails
  rspec_junit_formatter
  rubocop
  rubocop-rails
  sass-rails
  selenium-webdriver
  slack-incoming-webhooks
  spring
  spring-watcher-listen
  sprockets
  sprockets-rails
  tzinfo-data
  uglifier (>= 1.3.0)
  web-console
  webdrivers
  webpacker
  xmlrpc
  yaml_db

RUBY VERSION
   ruby 2.7.1p83

BUNDLED WITH
   2.1.4

```
</details>

**Comment by jonathanhefner** [comment]
@s-katoh I am still unable to reproduce this using Rails 6.0.3 and Ruby 2.7.1.  What command are you using to launch the console?

**Comment by eugeneius** [comment]
It's reproducible when the console uses pry instead of irb, as the Gemfile above does:

```ruby
gem 'pry-rails'
```

This was fixed in https://github.com/rails/rails/pull/37468, which will be released in 6.0.4.
